episodes/itksnap.Rmd

```diff
@@ -0,0 +1,630 @@
+Therefore, we are going to convert the image saved on disk to a floating point image. 
+We are going to use 32 bit rather than 64 bit floating point, as this is usually sufficiently accurate. 
+```
+print(nii3ct.get_data_dtype())
+#int16
+nii3ct.set_data_dtype(np.float32)
+print(nii3ct.get_data_dtype())
+#float32
+```
+
+We are also going to set the qform to unknown as suggested above.
+This also updates the corresponding fields in the header. 
+Use the `set_qform` function to set the `qform` code. 
+To just set the code and not modify the other `qform` values in the header, the first input should be None. 
+It is not necessary to modify the other `qform` values as they should be ignored when the code is set to unknown.
+```
+
+nii3ct.set_qform(None, code = 'unknown')
+print(nii3ct.header)
+...
+qform_code      : unknown
+...
+
+```
+
+The floating point image can now be written to disk using the save function. 
+Note. We do not need to manually modify the type of the image data – setting the data type for the Nifti1Image object tells it which type to use when writing the image data to disk.
+```
+nib.save(nii3ct, 'data/3_ct_lung__30__b31f_float32.nii.gz')
+```
+
+### 16-bit integers vs 32 bit floats
+
+The file size of `3_ct_lung__30__b31f_float32.nii.gz` is larger than  `3_ct_lung__30__b31f.nii.gz` (60 MB, 46 MB, respectively) because of the conversion from 16 bit integers to 32 bit floats. 
+Both images are compressed, but the compression is more efficient for the floating point image as it is not double than the original.
+
+* int16
+```
+nii3ct_int16 = nib.load("data/3_ct_lung__30__b31f.nii.gz")
+
+print(nii3ct_int16.get_data_dtype())
+#int16
+
+print(nii3ct_int16.header)
+#<class 'nibabel.nifti1.Nifti1Header'> object, endian='<'
+#sizeof_hdr      : 348
+#data_type       : b''
+#db_name         : b''
+#extents         : 0
+#session_error   : 0
+#regular         : b''
+#dim_info        : 0
+#dim             : [  3 512 512 175   1   1   1   1]
+#intent_p1       : 0.0
+#intent_p2       : 0.0
+#intent_p3       : 0.0
+#intent_code     : none
+#datatype        : int16
+#bitpix          : 16
+#...
+```
+* float32
+```
+nii3ct_float32 = nib.load("data/3_ct_lung__30__b31f_float32.nii.gz")
+
+print(nii3ct_float32.get_data_dtype())
+#float32
+
+print(nii3ct_float32.header)
+#<class 'nibabel.nifti1.Nifti1Header'> object, endian='<'
+#sizeof_hdr      : 348
+#data_type       : b''
+#db_name         : b''
+#extents         : 0
+#session_error   : 0
+#regular         : b''
+#dim_info        : 0
+#dim             : [  3 512 512 175   1   1   1   1]
+#intent_p1       : 0.0
+#intent_p2       : 0.0
+#intent_p3       : 0.0
+#intent_code     : none
+#datatype        : float32
+#bitpix          : 32
+#slice_start     : 0
+#...
+```
+
+
+
+### Cropping data
+
```

Comment:
Images often contain a lot of background voxels. While this informaiton can be useful in some instances, it often just takes up valuable RAM memory, especially for large images.

Reply:
[70bb35c](https://github.com/HealthBioscienceIDEAS/Medical-Image-Registration-Short-Course/pull/5/commits/70bb35cd32e535f1b403138a7f5f8b251cdbe5c0)